src/Input/Select/Select.tsx

```diff
@@ -15,346 +15,271 @@ import {
 
-  handleFocusOut = (
-    onBlur: (e: React.FocusEvent<HTMLInputElement>) => void
-  ) => {
-    const listener = (e: React.FocusEvent<HTMLInputElement>) => {
-      this.setState({
-        floating: e.target.value.length > 0,
-      });
+  React.useEffect(() => {
+    setFilterValue(React.Children.map(children, data => data));
+  }, [children, value]);
+
+  const handleFocusOut = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      setFloating(e.target.value.length > 0);
 
       if (onBlur !== undefined) {
         return onBlur(e);
       }
-    };
-
-    return listener;
-  };
+    },
+    []
+  );
 
-  handleFocus = (onFocus: (e: React.FocusEvent<HTMLInputElement>) => void) => {
-    const listener = (e: React.FocusEvent<HTMLInputElement>) => {
-      this.setState({
-        isFocus: true,
-      });
+  const handleFocus = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      setIsFocus(true);
 
       if (onFocus !== undefined) {
         return onFocus(e);
       }
-    };
-
-    return listener;
-  };
+    },
+    []
+  );
 
-  handleChange = (
-    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
-  ) => {
-    const listener = (e: React.ChangeEvent<HTMLInputElement>) => {
-      const { children } = this.props;
+  // Should be called when the user types into the input
+  const handleChange = React.useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
       const childrenArray = React.Children.toArray(children) as Array<
         React.ReactElement<SelectItemProps>
       >;
 
-      this.setState({
-        selectedValue: e.target.value,
-        filterValue: childrenArray.filter(data =>
+      setSelectedValue(e.target.value);
+      setFilterValue(
+        childrenArray.filter(data =>
           data.props.children
             .toLowerCase()
             .includes(e.target.value.toLowerCase())
-        ),
-        cursor: 0,
-      });
-
-      if (onChange !== undefined) {
-        return onChange(e);
-      }
-    };
-
-    return listener;
-  };
-
-  handleClick = (
-    onOptionClick: (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => void
-  ) => {
-    const listener = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
-      const { children, onChange } = this.props;
-      this.setState({
-        selectedValue: e.currentTarget.innerText,
-        filterValue: React.Children.map(children, data => data),
-        isFocus: false,
-      });
+        )
+      );
+      setCursor(0);
 
       if (onChange !== undefined) {
-        const element = e.target as HTMLElement;
-        onChange(element.dataset.value);
+        return onChange(e.target.value);
       }
+    },
+    [children, onChange]
+  );
+
+  // Should be called when the user selects an option
+  const handleSelect = React.useCallback(() => {
+    const activeElement = getActiveElement();
+    const selectedValue = activeElement.innerText;
+    setSelectedValue(selectedValue || '');
```

Comment:
Should we have an empty string as a fallback?

Reply:
Yeah this seems to have no effect, I'm removing it.